homeassistant/components/fibaro/sensor.py

```diff
@@ -144,8 +144,20 @@ def __init__(
 
     def update(self) -> None:
         """Update the state."""
-        with suppress(KeyError, ValueError):
-            self._attr_native_value = float(self.fibaro_device.properties.value)
+        with suppress(KeyError):
+            raw_value = self.fibaro_device.properties.value
+
+        if raw_value is None:
+            return raw_value
+
+        try:
+            # most sensors provide a number (HC2 as string, HC3 as float or int value)
+            self._attr_native_value = float(raw_value)
+        except (TypeError, ValueError):
+            # some sensors like accelerator provides a complex value,
+            # e.g. a accelerator provides {"x": 0.0, "y": 0.0, "z": 0.0}
+            # (HC2 as string HC3 as structure which is converted to RecursiveDict by the fiblary)
+            self._attr_native_value = str(raw_value)
```

Comment:
How is the dict with multiple items represented here?

Reply:
It depends on the HC version. For HC 3 the string value which is set as sensor value is like this `{"x": 0.0, "y": 0.0, "z": 0.0}` and the string value for HC 2 is like this `{"x":0.0000000000000000,"y":0.0000000000000000,"z":0.0000000000000000}`.

If your question is how the json sent from fibaro is, then for HC 3 it is `"value": {"x": 0.0, "y": 0.0, "z": 0.0}` and for HC 2 `"value": "{\"x\":0.0000000000000000,\"y\":0.0000000000000000,\"z\":0.0000000000000000}"`.

Reply:
Setting a stringified dict as state doesn't seem good. Each of the values should probably be its own sensor.

Reply:
I'm not sure if that helps, also the 3 values are highly related.
What about having a sum of the three values as number value so it can be used to detect a tampering by value > 0 and probably add x/y/z values as extra state attributes?
Otherwise I would suggest that we just not create entities from such sensors that we at least do not get errors in the log file.
Or third option is just to catch the error, this is what is happening anyway currently for HC2, an error is only thrown in case of HC3.

Reply:
Sum + attributes could be ok. If the measurement isn't interesting it's ok to not create sensors for it too.

Reply:
The 3 values represent relative movement in 3 dimensions , height width length . As @rappenze already said it can be used to detect tampering but it can also detect another movement like a earthquake . So I dont agree that this measurement isnt interesting.

Reply:
Ok so it looks like a better option is to have a binary sensor (this is as well used for accelerometer in other integrations).
So I changed the PR completely to add the accelerometer as binary sensor instead of a normal sensor and added the logic there.
Also I added the 3 values x / y / z as extra state attributes.